lucet-module-data/src/signature.rs

```diff
@@ -0,0 +1,172 @@
+        raw_module_and_data.patch_module_data(&cleared_module_data_bin);
+
+        minisign::verify(
+            &pk,
+            &signature_box,
+            Cursor::new(&raw_module_and_data.obj_bin),
+            true,
+            false,
+        )
+        .map_err(|e| ModuleSignatureError(e))
+    }
+}
+
+struct SymbolData {
+    offset: usize,
+    len: usize,
+}
+
+pub struct RawModuleAndData {
+    pub obj_bin: Vec<u8>,
+    pub module_data_offset: usize,
+    pub module_data_len: usize,
+}
+
+impl RawModuleAndData {
+    pub fn from_file<P: AsRef<Path>>(path: P) -> Result<Self, io::Error> {
+        let mut obj_bin: Vec<u8> = Vec::new();
+        File::open(&path)?.read_to_end(&mut obj_bin)?;
+
+        let module_data_symbol_data = Self::symbol_data(&obj_bin, "lucet_module_data", true)?
+            .ok_or(io::Error::new(
+                io::ErrorKind::InvalidInput,
+                "module data not found",
+            ))?;
+        let module_data_len_symbol_data =
+            Self::symbol_data(&obj_bin, "lucet_module_data_len", true)?.ok_or(io::Error::new(
+                io::ErrorKind::InvalidInput,
+                "module length not found",
+            ))?;
+        assert_eq!(module_data_len_symbol_data.len, 4);
+        assert_eq!(
+            LittleEndian::read_u32(&obj_bin[module_data_len_symbol_data.offset..]) as usize,
+            module_data_symbol_data.len
+        );
+        Ok(RawModuleAndData {
+            obj_bin,
+            module_data_offset: module_data_symbol_data.offset,
+            module_data_len: module_data_symbol_data.len,
+        })
+    }
+
+    pub fn module_data_bin(&self) -> &[u8] {
+        &self.obj_bin[self.module_data_offset as usize
+            ..self.module_data_offset as usize + self.module_data_len]
+    }
+
+    pub fn module_data_bin_mut(&mut self) -> &mut [u8] {
+        &mut self.obj_bin[self.module_data_offset as usize
+            ..self.module_data_offset as usize + self.module_data_len]
+    }
+
+    pub fn patch_module_data(&mut self, module_data_bin: &[u8]) {
+        self.module_data_bin_mut().copy_from_slice(&module_data_bin);
+    }
+
+    pub fn write_patched_module_data<P: AsRef<Path>>(
+        &self,
+        path: P,
+        patched_module_data_bin: &[u8],
+    ) -> Result<(), io::Error> {
+        let mut fp = OpenOptions::new()
+            .write(true)
+            .create_new(false)
+            .open(&path)?;
+        fp.seek(SeekFrom::Start(self.module_data_offset as u64))?;
+        fp.write_all(&patched_module_data_bin)?;
+        Ok(())
+    }
+
+    // Retrieving the offset of a symbol is not supported by the object crate.
+    // In Mach-O, actual file offsets are encoded, whereas Elf encodes virtual
+    // addresses, requiring extra steps to retrieve the section, its base
+    // address as well as the section offset.
+
+    // Elf
+    #[cfg(not(target_os = "macos"))]
```

Comment:
Can this be `#[cfg(all(target_family = "unix", not(target_os = "macos")))]` just to be a clear breakage point when building Lucet for a platform we know won't work? (Really just leaves Windows, but still!)

Reply:
Sure :)

Reply:
Done, we're now breaking on non-Unix systems. Although I hope we'll support WebAssembly at some point :)